src/persist/src/indexed/mod.rs

```diff
@@ -41,7 +41,68 @@ use crate::indexed::encoding::{
 };
 use crate::indexed::metrics::Metrics;
 use crate::pfuture::PFutureHandle;
-use crate::storage::{Blob, Log, SeqNo};
+use crate::storage::{Blob, BlobRead, Log, SeqNo};
+
+/// WIP
+#[derive(Debug)]
+pub enum CmdRead {
+    /// WIP
+    Load(String, (String, String), PFutureHandle<Option<Id>>),
+    /// Returns a [Description] of the stream identified by `id_str`.
+    //
+    // TODO: We might want to think about returning only the compaction frontier
+    // (since) and seal timestamp (upper) here. Description seems more oriented
+    // towards describing batches, and in our case the lower is always
+    // `Antichain::from_elem(Timestamp::minimum())`. We could return a tuple or
+    // create our own Description-like return type for this.
+    GetDescription(String, PFutureHandle<Description<u64>>),
+    /// Returns a [Snapshot] for the given id.
+    Snapshot(Id, PFutureHandle<ArrangementSnapshot>),
+    /// Registers a callback to be invoked on successful writes and seals.
+    ///
+    /// Also returns a copy of the snapshot so that users can, if they want,
+    /// apply their logic to a consistent read of the entire stream.
+    Listen(
+        Id,
+        ListenFn<Vec<u8>, Vec<u8>>,
+        PFutureHandle<ArrangementSnapshot>,
+    ),
+}
```

Comment:
imagine a Replay or Tail or whatever command here that takes a `Vec<ListenEvent<Vec<u8>, Vec<u8>>`

Reply:
do you imagine a world where we have that _and_ Listen or do you think that would replace Listen?

Reply:
they're basically opposites, so we'll need them both

Reply:
is the distinction tail/replay sends from one (writer) persist runtime -> another (reader) persist runtime in a different process, whereas listen is the way a persist runtime sends updates to some other thread / component that's not part of persist running on the same process?

I dunno they definitely seem different but it seems to me like they're more similar than not - like it seems like `Tail` is like `Listen`, but over the network and to different persist runtimes -- is that a fair way of looking at it?

Reply:
gentle ping on this question @danhhz